app/core/caching.py

```
@@ -0,0 +1,99 @@
+            # Generate cache key
+            cache_key = f"{key_prefix}:{func.__name__}"
+            
+            # Try to get from cache first
+            cached_result = redis_service.get(cache_key)
+            if cached_result is not None:
+                logger.info(f"Cache hit for key: {cache_key}")
+                return cached_result
+            
+            # Cache miss - execute function
+            logger.info(f"Cache miss for key: {cache_key}")
+            result = await func(*args, **kwargs) if hasattr(func, '__call__') else func(*args, **kwargs)
```

Ruff (0.13.1)

Using `hasattr(x, "__call__")` to test if x is callable is unreliable. Use `callable(x)` for consistent results.

Replace with `callable()`

(B004)